src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,133 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/** Contains interaction with user, meaning printing and reading through the scanner. */
+public class Ui {
+  Scanner scanner;
+
+  Ui() {
+    this.scanner = new Scanner(System.in);
+  }
+
+  public void showLine() {
+    String line = "-----------------------------------";
+    System.out.println(line);
+  }
+
+  public void showError(String errorMessage) {
+    showLine();
+    System.out.println(errorMessage);
+    showLine();
+  }
+
+  public void showWelcome() {
+    showLine();
+    String logo =
+            " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+    System.out.println("Hello from\n" + logo);
+    System.out.println("What can I do for you?");
+    showLine();
+  }
+
+  public void printWithFormat(String input, String type, TaskList tasks) {
+    switch (type) {
+    case "list":
```

Comment:
Good use of case instead of if-else statements as there are several different possibilities.